packages/core/src/tools/sanitize.ts

```diff
@@ -57,12 +58,12 @@ export function sanitize(source: unknown, maxCharacterCount = SANITIZE_DEFAULT_M
     containerQueue,
     visitedObjectsWithPath
   )
-  let accumulatedSize = JSON.stringify(sanitizedData)?.length || 0
-  if (accumulatedSize > maxCharacterCount) {
+  let accumulatedCharacterCount = JSON.stringify(sanitizedData)?.length || 0
+  if (isOverCharacterLimit(accumulatedCharacterCount, maxCharacterCount, true, source)) {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏using a boolean in this case is not really explicit when reading function call. What about passing the change type instead?

```ts
function isOverCharacterLimit(
  currentCharacterCount: number,
  maxCharacterCount: number,
  changeType: 'discarded' | 'truncated',
  source: unknown
) 
```

Reply:
👌